File: SchoolWebApp-API/Project.Core/Entities/CBE/Responsibilities/StudentResponsibility.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.Shared;
using SchoolWebApp.Core.Entities.Students;

namespace SchoolWebApp.Core.Entities.CBE.Responsibilities
{
    public class StudentResponsibility: Base
    {
        public int AcademicYearId { get; set; }
        public AcademicYear? AcademicYear { get; set; }

        public int StudentId { get; set; }
        public Student? Student { get; set; }

        public int ResponsibilitySocialSkillId { get; set; }

        public string? Description { get; set; } = null;
    }
}

File: SchoolWebApp-Client/src/app/cbe/responsibilities/models/student-responsibility.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class StudentResponsibility extends ResourceModel<StudentResponsibility> {
    public academicYearId?: number;
    public academicYear?: any;
    public studentId?: number;
    public student?: any;
    public responsibilitySocialSkillId?: number;
    public responsibilitySocialSkill?: any;
    public description?: string;

    constructor(model?: Partial<StudentResponsibility>) {
        super(model);
    }
}
